internal/controller/changetransferpolicy_controller.go

```diff
@@ -473,3 +486,36 @@ func (r *ChangeTransferPolicyReconciler) mergePullRequests(ctx context.Context,
 
 	return nil
 }
+
+func templatePullRequest(gitRepo *promoterv1alpha1.GitRepository, ctp *promoterv1alpha1.ChangeTransferPolicy) (string, string, error) {
+	titleTemplate := gitRepo.Spec.TitleTemplate
+	if titleTemplate == "" {
+		titleTemplate = DefaultPRTitleTemplate
+	}
+
+	tmpl, err := template.New("title").Funcs(sprig.FuncMap()).Parse(titleTemplate)
+	if err != nil {
+		return "", "", fmt.Errorf("failed to parse PR title template: %w", err)
+	}
+	var buf bytes.Buffer
+	if err := tmpl.Execute(&buf, ctp); err != nil {
+		return "", "", fmt.Errorf("failed to execute PR title template: %w", err)
+	}
+	title := buf.String()
+	buf.Reset()
+
+	bodyTemplate := gitRepo.Spec.BodyTemplate
+	if bodyTemplate == "" {
+		bodyTemplate = DefaultPRBodyTemplate
+	}
+	tmpl, err = template.New("body").Funcs(sprig.FuncMap()).Parse(bodyTemplate)
```

Comment:
Will probably want to pop `env`, `expandenv`, and `getHostByName` out of the function map, just to be a bit more secure (Argo CD, Events, and Workflows all do this as well).